src/Sulu/Bundle/HttpCacheBundle/CacheLifetime/CacheLifetimeRequestEnhancer.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) Sulu GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\HttpCacheBundle\CacheLifetime;
+
+use Symfony\Component\HttpFoundation\RequestStack;
+
+class CacheLifetimeRequestEnhancer
+{
+    const ATTRIBUTE_KEY = '_cacheLifetime';
+
+    /**
+     * @var RequestStack
+     */
+    private $requestStack;
+
+    public function __construct(RequestStack $requestStack)
+    {
+        $this->requestStack = $requestStack;
+    }
+
+    public function setCacheLifetime(int $cacheLifetime)
```

Comment:
should we call this `addMaxCacheLifetime`? think i would expect to overwrite the current value if it is called `setCacheLifetime `

Reply:
It *does* overwrite the current value, but only some check passes. Feels to me much more like a setter than an adder.